src/Workspaces/Remote/ServiceHub/Services/ConvertTupleToStructCodeRefactoringProvider/RemoteConvertTupleToStructCodeRefactoringService.cs

```diff
@@ -35,9 +35,8 @@ public RemoteConvertTupleToStructCodeRefactoringService(in ServiceConstructionAr
             bool isRecord,
             CancellationToken cancellationToken)
         {
-            return RunServiceAsync(async cancellationToken =>
+            return RunServiceWithSolutionAsync(solutionInfo, async solution =>
             {
-                var solution = await GetSolutionAsync(solutionInfo, cancellationToken).ConfigureAwait(false);
                 var document = solution.GetDocument(documentId);
```

Comment:
the new pattern for OOP features.  Features call RunServiceWithSolutionAsync passing in their PinnedSolutionInfo they are associated with.  They also pass in a callback which is given the solution that is computed once done.  This allows that computation to be shared among multiple concurrent callers into oop with teh same solution checksum.